### DISCUSSION TITLE
support of rar files

### DISCUSSION THREAD

**Discussion by mic62** [mucommander/mucommander#620] [Ideas]
I just noticed that muCommander won't open rar-files. How about supporting that file type directly or via a plugin?

**Comment by ahadas** [comment]
The RAR format is already supported [besides version 5](https://github.com/mucommander/mucommander/blob/master/package/readme.txt#L57). Did the archive you tried to browse appear empty?

**Comment by mic62** [comment]
When I doubleclick a rar.file I get this message:
Can't read folder.
Details:
java.lang.RuntimeException
 at com.mucommander.ui.event.LocationManager.setCurrentFolder(LocationManager.java:93)
 at com.mucommander.core.LocationChanger.setCurrentFolder(LocationChanger.java:349)
 at com.mucommander.core.BrowseLocationThread.run(BrowseLocationThread.java:365)

running muCommander on Windows 11:
muCommander
Version: 1.0.0-1
Build date: 2022/04/17
Build number: 15496dd

Java
Runtime version: 17.0.2
VM name: Java HotSpot(TM) 64-Bit Server VM
VM version: 17.0.2+8-LTS-86
VM vendor: Oracle Corporation

**Reply by ahadas** [reply]
I see, that stracktrace doesn't tell much, can you please attach the full log (located in ~/.mucommander/logs) and the rar file if possible?

**Reply by mic62** [reply]
There is no directory "Logs"
![image](https://user-images.githubusercontent.com/62608320/164891616-ac4ba132-2c48-4e5f-a10b-3bacd0fe37f3.png)

**Reply by ahadas** [reply]
Right, that's the installation folder - I was referring to the `.mucommander` under the home folder but that's ok, I see you added the file which is better

**Comment by mic62** [comment]
[V1.4_new_20220411.rar.txt](https://github.com/mucommander/mucommander/files/8546770/V1.4_new_20220411.rar.txt)
This is the file, I just had to rename to ....txt tu upload it here, as .rar files are not allowed. The file does open fine in 7zip.

**Reply by ahadas** [reply]
We get the following errors when trying to browse this file:
`2022-04-23 22:25:47.231 [Thread-60] WARN  com.github.junrar.Archive:352 - Support for rar version 5 is not yet implemented!
2022-04-23 22:25:47.231 [Thread-60] WARN  com.github.junrar.Archive:200 - exception in archive constructor maybe file is encrypted, corrupt or support not yet implemented`
And indeed this file was created with RAR 5 which is unfortunately not yet implemented by "unrar", the library we use for rar archives. There is an effort to implement version 5 there though (see https://github.com/junrar/junrar/issues/23) - we'll update when it's available (and regardless update to unrar v7.5.0)

**Reply by mic62** [reply]
Ah, I see. So I'll keep that in mind and hope that sometime in not too far future RAR 5 will be suppoprted.
Thanks for the info!
